pkg/frontend/querymiddleware/astmapper/instant_splitting_test.go

```diff
@@ -351,6 +351,80 @@ func TestInstantSplitterNoOp(t *testing.T) {
 	}
 }
 
+func TestUpdateRangeInterval(t *testing.T) {
+	tests := []struct {
+		expr         string
+		expectedExpr string
+		expectedErr  string
+	}{
+		{
+			expr:        `time()`,
+			expectedErr: "no matrix selector has been found",
+		}, {
+			expr:         `sum(rate(metric[1m]))`,
+			expectedExpr: `sum(rate(metric[1h]))`,
+		}, {
+			expr:         `sum(label_replace(rate(metric[1m]), "dst", "$1", "src", ".*"))`,
+			expectedExpr: `sum(label_replace(rate(metric[1h]), "dst", "$1", "src", ".*"))`,
+		}, {
+			expr:        `sum(rate(metric[1m])) + sum(rate(metric[5m]))`,
+			expectedErr: "multiple matrix selectors have been found",
+		},
+	}
+
+	for _, testData := range tests {
+		t.Run(testData.expr, func(t *testing.T) {
+			expr, err := parser.ParseExpr(testData.expr)
+			require.NoError(t, err)
+
+			actualErr := updateRangeInterval(expr, time.Hour)
+			if testData.expectedErr != "" {
+				require.Error(t, actualErr)
+				assert.Contains(t, actualErr.Error(), testData.expectedErr)
+			} else {
+				assert.Equal(t, testData.expectedExpr, expr.String())
+			}
+		})
+	}
+}
+
+func TestUpdateOffset(t *testing.T) {
+	tests := []struct {
+		expr         string
+		expectedExpr string
+		expectedErr  string
+	}{
+		{
+			expr:        `time()`,
+			expectedErr: "no vector selector has been found",
+		}, {
+			expr:         `sum(rate(metric[1m]))`,
+			expectedExpr: `sum(rate(metric[1m] offset 1h))`,
+		}, {
+			expr:         `sum(label_replace(rate(metric[1m]), "dst", "$1", "src", ".*"))`,
+			expectedExpr: `sum(label_replace(rate(metric[1m] offset 1h), "dst", "$1", "src", ".*"))`,
+		}, {
+			expr:        `sum(rate(metric[1m])) + sum(rate(metric[5m]))`,
+			expectedErr: "multiple vector selectors have been found",
+		},
+	}
+
+	for _, testData := range tests {
+		t.Run(testData.expr, func(t *testing.T) {
+			expr, err := parser.ParseExpr(testData.expr)
+			require.NoError(t, err)
+
+			actualErr := updateOffset(expr, time.Hour)
```

Comment:
And I would assume if we add a negative offset, it would be the same as no offset, right?
e.g., `sum(rate(metric[1m] offset 0m))` becomes `sum(rate(metric[1m]))

Reply:
>Can you also add a test with a negative offset?

Yep, good idea!

Reply:
>And I would assume if we add a zero offset, it would be the same as no offset, right?
That's correct. `time.Duration(0)` is also the golang zero value for the `OriginalOffset`.